packages/web-components/src/components/tree-view/tree-view.stories.ts

```diff
@@ -0,0 +1,348 @@
+          ${Document16({ slot: 'icon' })}
+        </cds-tree-node>
+        <cds-tree-node label="Business process mapping">
+          ${Document16({ slot: 'icon' })}
+        </cds-tree-node>
+      </cds-tree-node>
+      <cds-tree-node label="Business operations">
+        ${Document16({ slot: 'icon' })}
+      </cds-tree-node>
+      <cds-tree-node label="Cloud computing" is-expanded>
+        ${Folder16({ slot: 'icon' })}
+        <cds-tree-node label="Containers">
+          ${Document16({ slot: 'icon' })}
+        </cds-tree-node>
+        <cds-tree-node label="Databases">
+          ${Document16({ slot: 'icon' })}
+        </cds-tree-node>
+        <cds-tree-node label="DevOps">
+          ${Folder16({ slot: 'icon' })}
+          <cds-tree-node label="Solutions">
+            ${Document16({ slot: 'icon' })}
+          </cds-tree-node>
+          <cds-tree-node label="Case studies">
+            ${Folder16({ slot: 'icon' })}
+            <cds-tree-node label="Resources">
+              ${Document16({ slot: 'icon' })}
+            </cds-tree-node>
+          </cds-tree-node>
+        </cds-tree-node>
+      </cds-tree-node>
+      <cds-tree-node label="Data & Analytics" is-expanded>
+        ${Folder16({ slot: 'icon' })}
+        <cds-tree-node label="Big data">
+          ${Document16({ slot: 'icon' })}
+        </cds-tree-node>
+        <cds-tree-node label="Business Intelligence">
+          ${Document16({ slot: 'icon' })}
+        </cds-tree-node>
+      </cds-tree-node>
+      <cds-tree-node label="Models" is-expanded disabled>
+        ${Folder16({ slot: 'icon' })}
+        <cds-tree-node label="Audit">
+          ${Document16({ slot: 'icon' })}
+        </cds-tree-node>
+        <cds-tree-node label="Monthly data">
+          ${Document16({ slot: 'icon' })}
+        </cds-tree-node>
+        <cds-tree-node label="Data warehouse" is-expanded>
+          ${Folder16({ slot: 'icon' })}
+          <cds-tree-node label="Report samples">
+            ${Document16({ slot: 'icon' })}
+          </cds-tree-node>
+          <cds-tree-node label="Sales performance">
+            ${Document16({ slot: 'icon' })}
+          </cds-tree-node>
+        </cds-tree-node>
+      </cds-tree-node>
+    </cds-tree-view>
+  `,
+};
+
+export const withControlledExpansion = {
+  render: () => html`
+    <style>
+      ${styles}
+    </style>
+    <script>
+      function expandAll() {
+        document.querySelectorAll('cds-tree-node').forEach((node) => {
+          node.isExpanded = true;
+        });
+      }
+
+      function collapseAll() {
+        document.querySelectorAll('cds-tree-node').forEach((node) => {
+          node.isExpanded = false;
+        });
+      }
+    </script>
+    <div style="margin-bottom: 1rem">
+      <button type="button" onclick="expandAll()">Expand All</button>
+      <button type="button" onclick="collapseAll()">Collapse All</button>
+    </div>
+    <cds-tree-view label="Tree View">
```

Comment:
```suggestion
    <script>
      const tree = document.querySelector('cds-tree-view[controlled]');

      function expandAll() {
        tree.querySelectorAll('cds-tree-node').forEach((node) => {
          node.isExpanded = true;
        });
      }

      function collapseAll() {
        tree.querySelectorAll('cds-tree-node').forEach((node) => {
          node.isExpanded = false;
        });
      }
    </script>
    <div style="margin-block-end: 1rem">
      <cds-button onclick="expandAll()">Expand All</cds-button>
      <cds-button onclick="collapseAll()">Collapse All</cds-button>
    </div>

    <cds-tree-view controlled label="Tree View">
```

Just some small changes here so the toggle only affects this example and not the rest of the `treeview`s on the page